packages/playwright-core/src/server/browserType.ts

```diff
@@ -76,14 +77,28 @@ export abstract class BrowserType extends SdkObject {
 
   async _innerLaunchWithRetries(progress: Progress, options: types.LaunchOptions, persistent: types.BrowserContextOptions | undefined, protocolLogger: types.ProtocolLogger, userDataDir?: string): Promise<Browser> {
     try {
-      return this._innerLaunch(progress, options, persistent, protocolLogger, userDataDir);
+      return await this._innerLaunch(progress, options, persistent, protocolLogger, userDataDir);
     } catch (error) {
       // @see https://github.com/microsoft/playwright/issues/5214
-      const errorMessage = typeof error === 'object' && typeof error.message === 'string' ? error.message : '';
+      const errorMessage: string = typeof error === 'object' && typeof error.message === 'string' ? error.message : '';
       if (errorMessage.includes('Inconsistency detected by ld.so')) {
         progress.log(`<restarting browser due to hitting race condition in glibc>`);
         return this._innerLaunch(progress, options, persistent, protocolLogger, userDataDir);
       }
+      if (
+        // Chromium
+        errorMessage.includes('Missing X server') ||
```

Comment:
Shouldn't this all go to `_rewriteStartupError`?

Reply:
done